Tasks/Npm/npmtoolrunner.ts

```diff
@@ -74,4 +86,47 @@ export class NpmToolRunner extends tr.ToolRunner {
         let config = tl.execSync('npm', `config list ${this.dbg ? '-l' : ''}`, options);
         return options;
     }
+
+    private _getDebugLogPath(options?: tr.IExecSyncOptions): string {
+        // check cache
+        options.silent = true;
+        const cacheLocation = tl.execSync('npm', 'config get cache', options).stdout.trim();
```

Comment:
I moved this call to the constructor in order to avoid certain versions of npm overwriting the debug log when getting the cache location. I also prefer using execSync in this case to easily consume stdout. Thans for the context on exec vs execSync, I'll keep it in mind for any other uses.